util/blooms-db/src/db.rs

```diff
@@ -53,68 +63,121 @@ pub struct Database {
 	///
 	/// Every bloom is an ethereum header bloom
 	bot: File,
+}
+
+impl DatabaseFiles {
+	/// Open the blooms db files
+	pub fn open(path: &PathBuf) -> io::Result<DatabaseFiles> {
+		// let path = path.as_ref();
+		Ok(DatabaseFiles {
+			top: File::open(path.join("top.bdb"))?,
+			mid: File::open(path.join("mid.bdb"))?,
+			bot: File::open(path.join("bot.bdb"))?,
+		})
+	}
+
+	pub fn accrue_bloom(&mut self, pos: Positions, bloom: ethbloom::BloomRef) -> io::Result<()> {
+		self.top.accrue_bloom::<ethbloom::BloomRef>(pos.top, bloom)?;
+		self.mid.accrue_bloom::<ethbloom::BloomRef>(pos.mid, bloom)?;
+		self.bot.replace_bloom::<ethbloom::BloomRef>(pos.bot, bloom)?;
+		Ok(())
+	}
+
+	pub fn iterator_from(&mut self, pos: Positions) -> io::Result<DatabaseFilesIterator> {
```

Comment:
Just a curiosity: This method looks a little like the `std::io::Seek` trait. Would it be possible to refactor things so that `Positions` become `Position` and consist of single `u64` (which is then shifted properly to form the three variants needed for top, middle and bottom, kind of what happens in the `insert_blooms` method) and then impl `seek` for `DatabaseFilesIterator`? Not sure what we'd gain, and not in scope for this bug-fix PR but interested in your thoughts! :)

Reply:
I'm not actually quite sure how this `blooms-db` work so I'd say it's (as you said) out-of-scop for this PR, but @debris might have an opinion on this